WalletWasabi/BitcoinCore/Configuration/CoreConfig.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.BitcoinCore.Configuration
+{
+	public class CoreConfig
+	{
+		private List<CoreConfigLine> Lines { get; }
+
+		public int Count => Lines.Count;
+
+		public CoreConfig(CoreConfig coreConfig)
+		{
+			Lines = coreConfig.Lines.ToList();
+		}
+
+		public CoreConfig()
+		{
+			Lines = new List<CoreConfigLine>();
+		}
+
+		public IDictionary<string, string> ToDictionary()
+		{
+			var myDic = new Dictionary<string, string>();
+			foreach (var line in Lines)
+			{
+				if (line.HasKeyValuePair)
+				{
+					myDic.Add(line.Key, line.Value);
+				}
+			}
+
+			return myDic;
+		}
+
+		public override string ToString() => $"{Guard.Correct(string.Join(Environment.NewLine, Lines))}{Environment.NewLine}"; // Good practice to end with a newline.
+
+		/// <summary>
+		/// TryAdd and AddOrUpdate are sisters. TryAdd always considers the first occurrence of a key as valid, while AddOrUpdate considers the last occurrence of it.
+		/// </summary>
+		public void TryAdd(string configString)
+		{
+			foreach (var line in ParseConfigLines(configString))
+			{
+				if (line.HasKeyValuePair)
+				{
+					var foundLine = Lines.FirstOrDefault(x => x.Key == line.Key);
+					if (foundLine is null)
+					{
+						Lines.Add(line);
+					}
+				}
+				else
+				{
+					Lines.Add(line);
+				}
+				RemoveFirstEmptyDuplication();
+			}
+		}
+
+		/// <summary>
+		/// TryAdd and AddOrUpdate are sisters. TryAdd always considers the first occurrence of a key as valid, while AddOrUpdate considers the last occurrence of it.
+		/// </summary>
+		public bool AddOrUpdate(string configString)
+		{
+			var ret = false;
+			foreach (var line in ParseConfigLines(configString))
+			{
+				if (line.HasKeyValuePair)
+				{
+					var foundLine = Lines.FirstOrDefault(x => x.Key == line.Key);
+					if (foundLine is null)
+					{
+						Lines.Add(line);
+						ret = true;
+					}
+					else if (foundLine.Value != line.Value)
+					{
+						Lines.Remove(foundLine);
+						Lines.Add(line);
+						ret = true;
+					}
+				}
+				else
+				{
+					Lines.Add(line);
+					ret = true;
+				}
```

Comment:
Shouldn't we simply write it as follows:
```
if (line.HasKeyValuePair)
{
	var foundLine = Lines.FirstOrDefault(x => x.Key == line.Key);
	if (foundLine.Value != line.Value)
	{
		Lines.Remove(foundLine);
	}
}
Lines.Add(line);
ret = true;
```

Reply:
No, your code always adds the line. Currently:

```
if (foundLine is null) // add line 
else if (foundLine.Value != line.Value) // add line
else // don't add line
```

Reply:
Oh I missed that. Thanks